lib/season_statistics.rb

```diff
@@ -9,14 +9,125 @@ def biggest_bust(season)
       ps_game = games.find_all do |game|
         @games[game.game_id].type == "Postseason"
       end
-    require'pry';binding.pry
     end
   end
-end
 
-    # post_season_games = games_by_season[season].find_all do |game|
-    #   game.season == "Postseason"
-    # end
-    # regular_season_games = games_by_season[season].find_all do |game|
-    #   game.season == "Regular Season"
-    # end
+  def most_tackles(season_id)
+    filtered_games = @game_teams.find_all do |game|
+      @games[game.game_id].season == season_id
+    end
+
+    games_by_teams = filtered_games.group_by(&:team_id)
+
+    most_tackles_team_id = nil
+    most_tackles = 0
+    games_by_teams.each do |team_id, games_arr|
+      tackles = games_arr.sum {|game| game.tackles}
+      if tackles > most_tackles
+        most_tackles_team_id = team_id
+        most_tackles = tackles
+      end
+    end
+    @teams[most_tackles_team_id].team_name
+  end
+
+  def fewest_tackles(season_id)
+    filtered_games = @game_teams.find_all do |game|
+      @games[game.game_id].season == season_id
+    end
+
+    games_by_teams = filtered_games.group_by(&:team_id)
+
+    fewest_tackles_team_id = nil
+    fewest_tackles = 100_000
+    games_by_teams.each do |team_id, games_arr|
+      tackles = games_arr.sum {|game| game.tackles}
+      if tackles < fewest_tackles
+        fewest_tackles_team_id = team_id
+        fewest_tackles = tackles
+      end
+    end
+    @teams[fewest_tackles_team_id].team_name
+  end
+
+  def least_accurate_team(season_id)
+    filtered_games = @game_teams.find_all do |game|
+      @games[game.game_id].season == season_id
+    end
+
+    games_by_teams = filtered_games.group_by(&:team_id)
+
+    lowest_ratio_team_id = nil
+    lowest_ratio = 10
+    games_by_teams.each do |team_id, games_arr|
+      shot_ratio = games_arr.sum(&:goals).to_f / games_arr.sum(&:shots)
+      # require 'pry'; binding.pry
```

Comment:
Remove this pry

Reply:
I agree that I can see us creating a helper method during refactor tomorrow to include filtered games as it is repeated in most methods.